src/app/components/graph/graph.component.ts

```diff
@@ -73,62 +78,58 @@ export class GraphComponent implements OnInit, OnChanges {
-      hexToRgba(this.getColor(8), 60),
-    ],
-    borderColor: [
-      this.getColor(0),
-      this.getColor(1),
-      this.getColor(2),
-      this.getColor(3),
-      this.getColor(4),
-      this.getColor(5),
-      this.getColor(6),
-      this.getColor(7),
-      this.getColor(8),
-    ],
+    backgroundColor: [],
+    borderColor: [],
     borderWidth: 1
   }];
 
-  barColors = [
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(0), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(1), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(2), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(3), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(4), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(5), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(6), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(7), 60)},
-    {borderWidth: 1, backgroundColor: hexToRgba(this.getColor(8), 60)},
-  ];
+  barColors = [];
 
   legend = true;
 
   constructor() {
   }
 
   ngOnInit() {
+    let iterableColorList = this.generateIterableArray(this.colorList);
+    this.doughnutPolarColors[0].backgroundColor.push(
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+      hexToRgba(iterableColorList.next(), 60),
+    )
+    
+    iterableColorList = this.generateIterableArray(this.colorList);
+    this.doughnutPolarColors[0].borderColor.push(
+      iterableColorList.next(),
+      iterableColorList.next(),
+      iterableColorList.next(),
+      iterableColorList.next(),
+      iterableColorList.next(),
+      iterableColorList.next(),
+      iterableColorList.next(),
+      iterableColorList.next(),
+      iterableColorList.next(),
+    )
+
+    iterableColorList = this.generateIterableArray(this.colorList);
+    this.barColors.push(
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+      {borderWidth: 1, backgroundColor: hexToRgba(iterableColorList.next(), 60)},
+    )
```

Comment:
> Try to replace those hard coded initializers with a more dynamic approach using loops for example, without the need to repeat it manually for each color....
It's clear. I will make it.

> Additionaly you can fill all these arrays at the same time, without the need to re-initialize
I could not really catch it. What should the current method do? Just simply reset the counter? It would be really not bad, but then I would like to suggest the method name reset(). Or do you mean something else?

Reply:
I pushed a solution. I think it really should be better now

Reply:
The line `iterableColorList = this.generateIterableArray(this.colorList);` is called before each array you fill, then you iterate over this list by calling your `.next()` function 9 times for each of the 3 arrays you fill.

This fills all the three arrays with the same order of colors. 
To not rebuilt the iterableColorList before each array you fill you could add a `current()` method to the IterableArray, which looks something like this:
```
...
},
current(){
  return this.arr[this.current];
}
...
```

This would then allow to replace the reinitalization with something like this for example (in pseudo-code):

```
iterableColorList = this.generateIterableArray(this.colorList);
for....{
      this.doughnutPolarColors[0].backgroundColor.push(  [...]iterableColorList.next()[...]);
      this.doughnutPolarColors[0].borderColor.push([...]iterableColorList.current()[...]);
      this.barColors.push([...] iterableColorList.current() [...]);
}
```

I hope its clear what I mean, if not just ask again.

Reply:
yes, it's clear. Just 3 points to mention:
- the method name _current()_ already exists, I used the name _lastUsed()_
- the truth is the old variable _current_  was referred to the next value. I renamed it now.
- then the _lastUsed()_ method returns _this.arr[this.nextIndex-1];_ and there is an extra case if the _nextValue == 0_

I hope it's okay. Thank you for your suggestions.

Reply:
yes, it's clear. Just 3 points to mention:

the method name current() already exists, I used the name lastUsed()
the truth is the old variable current was referred to the next value. I renamed it now.
then the lastUsed() method returns this.arr[this.nextIndex-1]; and there is an extra case if the nextValue == 0
I hope it's okay. Thank you for your suggestions.